components/view_image/view_image.jsx

```diff
@@ -117,19 +117,34 @@ export default class ViewImageModal extends React.PureComponent {
         }
     }
 
-    UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
-        if (nextProps.show === true && this.props.show === false) {
-            this.onModalShown(nextProps);
-        } else if (nextProps.show === false && this.props.show === true) {
-            this.onModalHidden();
+    getSnapshotBeforeUpdate(prevProps) {
+        if (this.props.show === true && prevProps.show === false) {
+            return 'modalShown';
+        } else if (this.props.show === false && prevProps.show === true) {
+            return 'modalHidden';
         }
+        return null;
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if (snapshot !== null) {
```

Comment:
Instead of using `getSnapshotBeforeUpdate`, could you just compare `this.props.show` and `prevProps.show` here?

Reply:
Alright...I totally agree. I'll make the changes.